aws/src/main/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactory.java

```diff
@@ -121,7 +112,7 @@ protected <T extends AwsClientBuilder & AwsSyncClientBuilder> T configure(T clie
             .build());
 
     clientBuilder.region(Region.of(region));
-    clientBuilder.httpClientBuilder(AwsClientFactories.configureHttpClientBuilder(httpClientType));
+    awsProperties.applyHttpClientConfiguration(clientBuilder);
```

Comment:
Looks like we no longer need this `configure` method with your change. The `AssumeRoleRequest` can be created upfront, there is no need to create a new request every time the client is configured.

Reply:
Would it be better if we just created the `AssumeRoleRequest` upfront while still keeping this `configure` method? Since the four aws clients need some common configuration in the `AssumeRole` case, the `configure` method may help us reduce duplicated codes above.